Random.py

```diff
@@ -10,40 +9,58 @@ def __init__(self):
         self.students = {}
 
     def add_student(self, name, age, student_id):
-        if student_id in self.students:
+        # Error: Incorrect condition, should check if student_id in self.students
+        if student_id not in self.students:  
             print(f"Error: A student with ID {student_id} already exists.")
```

Comment:
There's an error in the condition checking for existing students. It should raise an error message if the student already exists.
```diff
-        if student_id not in self.students:
+        if student_id in self.students:
```
---